hudi-gcp/src/main/java/org/apache/hudi/gcp/bigquery/BigQuerySyncConfig.java

```diff
@@ -128,6 +128,12 @@ public class BigQuerySyncConfig extends HoodieSyncConfig implements Serializable
       .withDocumentation("Assume standard yyyy/mm/dd partitioning, this"
           + " exists to support backward compatibility. If you use hoodie 0.3.x, do not set this parameter");
 
+  public static final ConfigProperty<Boolean> BIGQUERY_SYNC_ALLOW_READ_OPTIMIZED_SYNC = ConfigProperty
+      .key("hoodie.gcp.bigquery.sync.allow_read_optimized_sync")
+      .defaultValue(false)
```

Comment:
Shall we enable by default? For other engines, when we started, we supported COW and MOR read optimized by default.
Also, let's track a ticket to deprecate this config once snapshot query is also supported for MOR.

Reply:
The main reason why I didn't have it enabled by default was since this is explicitly not allowed in the code today so I wanted to be on the safe side. I don't think there is any plan for Google Cloud to support snapshot query, have you heard otherwise?

Reply:
Let's remove the flag. Don't even need to log a warning. We can call out in the docs that RO queries will be supported for BQ.

Reply:
yeah i Agree, we should just call it out in the docs. Otherwise lgtm, @codope you can help merge when ready